### PR TITLE
cookbook article: windows development

### PR DESCRIPTION
**Pull request by boutell**
## Summary

Document how to get started with Apostrophe under Windows Subsystem for Linux. This process was tested on an older-generation i5 Windows 10 machine. `npm install` of `a3-boilerplate` took 7 minutes, even with an SSD. Daaang!

## What are the specific steps to test this change?

Ideally: try out the tutorial on a Windows machine. However getting this published for our customers who need to get their Windows developers to a better place is more important than waiting to get hold of one if you don't have one handy.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/cookbook/README.md [hunk 1]
```diff
@@ -8,4 +8,8 @@ Welcome to the cookbook. These code recipes show how to execute specific tasks,
 
 ## Hosting
```
Comment (BoDonkey) [inline docs/cookbook/README.md:9]: Based on the build, I believe this same info must also be added to the `.vuepress/sidebar.js` file in order to populate the page link in the documentation.

File: docs/cookbook/windows-development.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+::: warning
+We recommend that you use WSL2. If you have WSL1, here are [Microsoft's upgrade instructions.](https://docs.microsoft.com/en-us/windows/wsl/install#upgrade-version-from-wsl-1-to-wsl-2) As described in that article, you may need to take care of several steps including a kernel update and enabling virtualization in your BIOS. The correct steps to enable virtualization depend on your machine.
+
+If you have never installed WSL before, WSL2 will be the default. The rest of this article assumes a first-time install of WSL2.
+:::
+
+First, you must [install WSL according to the documentation](https://docs.microsoft.com/en-us/windows/wsl/install). In particular we strongly recommend that you [install Ubuntu 20.04 LTS, which can also be done from the Windows app store](https://apps.microsoft.com/store/detail/ubuntu-2004/9N6SVWS3RX71?hl=en-us&gl=US). This method was tested for this article. Newer versions of Ubuntu might not support everything covered here, and 20.04 is supported until 2030.
```
Comment (BoDonkey) [inline docs/cookbook/windows-development.md:13]: Ubuntu 20.04 can also be installed using `wsl --install -d Ubuntu-20.04`

File: docs/cookbook/windows-development.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+Second, launch Ubuntu 20.04 from the Start menu to access the Linux shell prompt. If you did not install Ubuntu via the Windows Store, you may need to access the prompt a different way, for instance by launching Powershell and typing `wsl`.
```
Comment (BoDonkey) [inline docs/cookbook/windows-development.md:15]: ~~may~~ might
This is to aggree with "If you did..."

**Comment by boutell** [comment]
Good catch. I also fixed the opposite problem for the profiling article.

**Comment by boutell** [comment]
Oops I think you are not done reviewing... let me know when you're ready for me to respond.

File: docs/cookbook/windows-development.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+::: warning
+If the command prompt does return, and you see a message like `fatal assertion`, then `mongod` was unable to start. Most likely you previously tried to run MongoDB in another way, and you need to fix a permissions problem and try again, like this:
+
+```bash
+sudo rm /tmp/mongodb-27017.sock
+mongod --dbpath=/home/apostrophe/mongodb-data/5.0
+```
+
+You should only have to do this once to clean up the mess. In future, **just remember: don't use `sudo`, you don't need it and it only makes a mess.** If there's an exception we'll explicitly show that in our tutorials.
```
Comment (BoDonkey) [inline docs/cookbook/windows-development.md:119]: "In the future..."

**Review by BoDonkey — CHANGES_REQUESTED** [review]
Minor editorial changes that you can choose to mostly ignore. The instructions work when running in a fresh Ubuntu 20.04 install. My Win 11 install borked for unrelated reasons, but later tonight I will try a run-through in a Bootcamp Win 10 install.

**Review by BoDonkey — APPROVED** [review]

**Pull request merged**